un_sdg/clean.py

```diff
@@ -0,0 +1,227 @@
+import shutil
+from datetime import datetime
+import json
+import numpy as np
+import re
+
+from pathlib import Path
+from tqdm import tqdm
+from typing import List, Tuple, Dict
+
+#from db import connection
+#from db_utils import DBUtils
+from un_sdg import (
+    INFILE,
+    ENTFILE,
+    OUTPATH,
+    DATASET_NAME,
+    DATASET_AUTHORS,
+    DATASET_VERSION
+)
+
+from un_sdg.core import (
+    create_short_unit,
+    extract_datapoints,
+    get_distinct_entities,
+    clean_datasets,
+    dimensions_description,
+    attributes_description,
+    create_short_unit,
+    get_series_with_relevant_dimensions,
+    generate_tables_for_indicator_and_series,
+    str_to_float
+)
+
+
+
+def load_and_clean():
+    # Load and clean the data
+    print("Reading in original data...") 
+    original_df = pd.read_csv(
+        INFILE, 
+        converters={'Value': str_to_float},
+        low_memory=False
+    )
+    original_df = original_df[original_df['Value'].notnull()]
+    print("Extracting unique entities to " + ENTFILE + "...")
+    original_df[['GeoAreaName']].drop_duplicates() \
+                                .dropna() \
+                                .rename(columns={'GeoAreaName': 'Country'}) \
+                                .to_csv(ENTFILE, index=False)
+    # Make the datapoints folder
+    Path(OUTPATH, 'datapoints').mkdir(parents=True, exist_ok=True)
+    return original_df
+
+"""
+Now use the country standardiser tool to standardise $ENTFILE
+1. Open the OWID Country Standardizer Tool
+   (https://owid.cloud/admin/standardize);
+2. Change the "Input Format" field to "Non-Standard Country Name";
+3. Change the "Output Format" field to "Our World In Data Name"; 
+4. In the "Choose CSV file" field, upload {outfpath};
+5. For any country codes that do NOT get matched, enter a custom name on
+   the webpage (in the "Or enter a Custom Name" table column);
+    * NOTE: For this dataset, you will most likely need to enter custom
+      names for regions/continents (e.g. "Arab World", "Lower middle
+      income");
+6. Click the "Download csv" button;
+7. Replace {outfpath} with the downloaded CSV;
+8. Rename the "Country" column to "country_code".
+"""
+
+### Datasets
+def create_datasets():
+    df_datasets = clean_datasets(DATASET_NAME, DATASET_AUTHORS, DATASET_VERSION)
+    assert df_datasets.shape[0] == 1, f"Only expected one dataset in {os.path.join(OUTPATH, 'datasets.csv')}."
+    print("Creating datasets csv...")
+    df_datasets.to_csv(os.path.join(OUTPATH, 'datasets.csv'), index=False)
+    return df_datasets
+
+### Sources
+
+def create_sources(original_df, df_datasets):
+    df_sources = pd.DataFrame(columns=['id', 'name', 'description', 'dataset_id'])
+    source_description_template = {
+        'dataPublishedBy': "United Nations Statistics Division",
+        'dataPublisherSource': None,
+        'link': "https://unstats.un.org/sdgs/indicators/database/",
+        'retrievedDate': datetime.now().strftime("%d-%B-%y"),
```

Comment:
Ah good point, I've updated it [here](https://github.com/owid/importers/commit/906998fa0d4c1cf0c538dc3f6c6f802f166092a6) so that DATASET_RETRIEVED_DATE is now the date the raw data file was last modified according to os.stat.